Implemented offset before stimulus

rickdott opened the pull request:
Change to allow adding additional samples before offset, in my usecase I later remove them from the dataset so they are not used in actual HMP estimation but I dont see why they couldn't be (perhaps when you want to examine prestim effects like anticipation)

Implementation is very simple, @GWeindel do you think this might mess with RT calculation somewhere? If you do RT * sfreq to find the samples it should now be shifted by offset_before, what do you think?

Also implemented a check for when selected time before stim is earlier than tmin of the epochs object, assuming this is **always** in seconds

GWeindel commented on hmp/preprocessing.py:
```diff
@@ -358,6 +360,7 @@ def __init__(  # noqa  # Might need to be refactored.
 
         data.attrs["sfreq"] = sfreq
         data.attrs["offset"] = offset
```
Change t
```suggestion
        data.attrs["offset_end"] = offset_end
```

rickdott replied:
I kept the pre-existing offset as 'offset' to not break any other code that uses this term (including my own). Do you prefer changing this? We could also specify that 'offset' is at the end in the docstring. Let me know which you prefer and I'll adjust accordingly.

GWeindel commented on hmp/preprocessing.py:
```diff
@@ -358,6 +360,7 @@ def __init__(  # noqa  # Might need to be refactored.
 
         data.attrs["sfreq"] = sfreq
         data.attrs["offset"] = offset
+        data.attrs["offset_before"] = offset_before
```
Clearer I would say, also for people who might use other things than stimulus and response as interval
```suggestion
        data.attrs["offset_start"] = offset_start
```

GWeindel commented on hmp/io.py:
```diff
@@ -642,7 +656,9 @@ def hmp_data_format(
-        Offset in seconds to apply to the data.
+        Offset in samples to apply after the response time.
+    offset_before : float, default=0
+        Offset in samples to apply before the stimulus onset.
```
Is this negative then if we want samples before stimulus?

rickdott commented:
Adjusted naming, also looked for '.offset' in the HMP repo and adjusted where necessary. Name is now 'offset_end'. Tested by fitting a small HMP and visualizing it.

@GWeindel do you want me to adjust the user-facing (read_mne_data) parameter 'offset_start' to use negative numbers? I think using negative numbers now would shorten the epoch.

GWeindel commented:
This looks good, I fear though that the structure changes in #247 break your changes. Would you consider applying this to #247 after it has been merged (matter of days)? In this new version you don't need to change the input functions and instead only change the transformer operations (yes I insist on that name :))

rickdott commented:
Of course!
